### ISSUE TITLE
Callback does only fire when function is anonymous

### ISSUE THREAD

**Issue by HendrikF** [t4t5/sweetalert#431]
I experienced problems with the callback and found these lines:

``` javascript
/*
 *  User clicked on "Cancel"
 */
var handleCancel = function handleCancel(modal, params) {
  // Check if callback function expects a parameter (to track cancel actions)
  var functionAsStr = String(params.doneFunction).replace(/\s/g, '');
  var functionHandlesCancel = functionAsStr.substring(0, 9) === 'function(' && functionAsStr.substring(9, 10) !== ')';

  if (functionHandlesCancel) {
    params.doneFunction(false);
  }

  if (params.closeOnCancel) {
    sweetAlert.close();
  }
};
```

The check for `functionHandlesCancel` assumes, that the given callback is anonymous.

In my case we track frontend errors with angular and stacktrace.js. We give every function a name to make debugging easier.

If the function is named, the `functionAsString` contains something like `functioncallback(isConfirm)`.

Greets, Hendrik.

**Comment by HendrikF** [comment]
typo :+1:

**Comment by taylortom** [comment]
+1 for this

**Comment by t4t5** [comment]
Fixed in 2.0 with the help of promises.
